lib/tasks/upgrade.rake

```diff
@@ -1,6 +1,14 @@
 require 'set'
 namespace :upgrade do
 
+  desc "Ensure all section numbers are unique per Phase."
+  task clean_section_numbers: :environment do
+    Phase.all.each do |phase|
+      ids_in_order = phase.sections.order("number, created_at").pluck(:id)
```

Comment:
created_at seems like as sensible a way as any to determine the order if multiple have the same number.